src/lib/fuquAdapter.ts

```diff
@@ -0,0 +1,82 @@
+import { BareEvent, Event, FinishedMessageMetadata, FuQu, FuQuOptions, Handler, IncomingMessageMetadata } from './fuqu';
+
+export type FuQuCreator<O extends FuQuOptions<any, any>, Message> = <
+    Payload extends object,
+    Attributes extends { [key: string]: string } = Record<any, any>
+>(
+    client: any,
+    topicName: string,
+    options?: O
+) => FuQu<Payload, Attributes, Message>;
+
+export interface FuQuAdapter<P, A, M> {
+    name: string;
+    publishJson: (payload: P, attributes?: A) => Promise<void>;
+    registerHandler: (handler: Handler<P, A, M>) => Promise<void> | void;
+    ack: (message: M) => Promise<void> | void;
+    nack: (message: M) => Promise<void> | void;
+    close: () => Promise<void>;
+    isAlive: () => Promise<boolean>;
+    createIncomingMessageMetadata: (message: M, payload: P) => IncomingMessageMetadata<P, A>;
+    createFinishedMessageMetadata: (
+        incomingMetadata: IncomingMessageMetadata<P, A>,
+        message: M
+    ) => FinishedMessageMetadata<P, A>;
+}
+
+export const createFuQu = <P, A, M>(
+    adapter: FuQuAdapter<P, A, M>,
+    topicName: string,
+    options?: FuQuOptions
+): FuQu<P, A, M> => {
+    const debugLog = require('debug')(`fuqu:${topicName}`);
```

Comment:
Why inline require?

Reply:
I think it was lacking types. I'll see what can be done about it 👍

Reply:
Fixed. I installed the types and used import statement. My bad, I added `debug` hastily and was just lazy at the moment. Thanks :cat: